health/state.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package health
+
+import (
+	"time"
+)
+
+// State is a representation of the health status of the backend, which is
+// provided to the client UI via LocalAPI. It is used to simplify the data structure
+// provided over JSON.
+type State struct {
+	// Warnings maps each WarnableCode to a WarningStateRepresentation.
+	Warnings map[WarnableCode]WarningStateRepresentation
+}
+
+type WarningStateRepresentation struct {
+	WarnableCode WarnableCode
+	Severity     Severity
+	Title        string
+	Text         string
+	BrokenSince  time.Time
+	Args         map[string]string
+}
+
+func (t *Tracker) CurrentState() *State {
+	if t.nil() {
+		return nil
+	}
+
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	wm := map[WarnableCode]WarningStateRepresentation{}
```

Comment:
do you need to even make these if there's nothing in warning state? let it stay nil?

Reply:
Or: 

```
if len(t.warnableVal) == 0 {
    return nil
```

before this?

And document when CurrentState return snil? And what State.Warnings of nil means? I guess it should never be nil because the outer one is only sent if there are warnings?